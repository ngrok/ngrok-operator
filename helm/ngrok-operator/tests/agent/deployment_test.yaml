suite: test agent deployment
templates:
- agent/deployment.yaml
- agent/rbac.yaml
set:
  ingress:
    enabled: true
tests:
- it: Should match snapshot
  asserts:
  - matchSnapshot: {}
- it: Adds .Values.podLabels to the controller deployment podspec
  set:
    podLabels:
      labelKey1: labelValue1
      labelKey2: labelValue2
  template: agent/deployment.yaml
  asserts:
  - isSubset:
      path: spec.template.metadata.labels
      content:
        labelKey1: labelValue1
        labelKey2: labelValue2
- it: Does not add .Values.podLabels to the controller deployment's selector
  set:
    podLabels:
      labelKey1: labelValue1
      labelKey2: labelValue2
  template: agent/deployment.yaml
  asserts:
  - isNotSubset:
      path: spec.selector.matchLabels
      content:
        labelKey1: labelValue1
        labelKey2: labelValue2
- it: Supports selecting which nodes the agent should run on with node selectors
  set:
    agent:
      nodeSelector: &nodeSelector
        disktype: ssd
  template: agent/deployment.yaml
  asserts:
  - equal:
      path: spec.template.spec.nodeSelector
      value: *nodeSelector
- it: Supports setting tolerations on the agent pods
  set:
    agent:
      tolerations:
      - &toleration
        key: "key1"
        operator: "Equal"
        value: "value1"
        effect: "NoSchedule"
  template: agent/deployment.yaml
  asserts:
  - contains:
      path: spec.template.spec.tolerations
      content: *toleration
- it: Supports setting topologySpreadConstraints on the agent pods
  set:
    agent:
      topologySpreadConstraints:
      - &tsc
        maxSkew: 1
        topologyKey: kubernetes.io/hostname
        whenUnsatisfiable: DoNotSchedule
        labelSelector:
          matchLabels:
            app: foo
        matchLabelKeys:
          - pod-template-hash
  template: agent/deployment.yaml
  asserts:
  - contains:
      path: spec.template.spec.topologySpreadConstraints
      content: *tsc
- it: Supports setting resources on the agent pods
  set:
    agent:
      resources: &resources
        requests:
          memory: "128Mi"
          cpu: "500m"
        limits:
          memory: "256Mi"
          cpu: "1000m"
  template: agent/deployment.yaml
  asserts:
  - equal:
      path: spec.template.spec.containers[0].resources
      value: *resources
