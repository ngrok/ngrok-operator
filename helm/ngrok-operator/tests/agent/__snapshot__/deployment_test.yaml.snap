Should match snapshot:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      annotations:
        checksum/rbac: 4b1abc17af462422ef4fed990f96e966defbc3047d301093cbf5b685c3036aae
      labels:
        app.kubernetes.io/component: agent
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: ngrok-operator
        app.kubernetes.io/part-of: ngrok-operator
        app.kubernetes.io/version: 0.14.3
        helm.sh/chart: ngrok-operator-0.17.0-rc.4
      name: RELEASE-NAME-ngrok-operator-agent
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/component: agent
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: ngrok-operator
      template:
        metadata:
          annotations:
            checksum/rbac: 4b1abc17af462422ef4fed990f96e966defbc3047d301093cbf5b685c3036aae
            prometheus.io/path: /metrics
            prometheus.io/port: "8080"
            prometheus.io/scrape: "true"
          labels:
            app.kubernetes.io/component: agent
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: ngrok-operator
        spec:
          affinity:
            nodeAffinity: null
            podAffinity: null
            podAntiAffinity:
              preferredDuringSchedulingIgnoredDuringExecution:
                - podAffinityTerm:
                    labelSelector:
                      matchLabels:
                        app.kubernetes.io/component: agent
                        app.kubernetes.io/instance: RELEASE-NAME
                        app.kubernetes.io/name: ngrok-operator
                    topologyKey: kubernetes.io/hostname
                  weight: 1
          containers:
            - args:
                - --enable-feature-ingress=true
                - --enable-feature-gateway=false
                - --description="The official ngrok Kubernetes Operator."
                - --zap-log-level=info
                - --zap-stacktrace-level=error
                - --zap-encoder=json
                - --health-probe-bind-address=:8081
                - --metrics-bind-address=:8080
                - --manager-name=RELEASE-NAME-ngrok-operator-agent-manager
              command:
                - /agent-manager
              env:
                - name: NGROK_AUTHTOKEN
                  valueFrom:
                    secretKeyRef:
                      key: AUTHTOKEN
                      name: RELEASE-NAME-ngrok-operator-credentials
                - name: POD_NAMESPACE
                  valueFrom:
                    fieldRef:
                      fieldPath: metadata.namespace
                - name: HELM_RELEASE_NAME
                  value: RELEASE-NAME
              image: docker.io/ngrok/ngrok-operator:0.14.3
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /healthz
                  port: 8081
                initialDelaySeconds: 15
                periodSeconds: 20
              name: agent
              readinessProbe:
                httpGet:
                  path: /readyz
                  port: 8081
                initialDelaySeconds: 5
                periodSeconds: 10
              resources:
                limits: {}
                requests: {}
              securityContext:
                allowPrivilegeEscalation: false
          serviceAccountName: RELEASE-NAME-ngrok-operator-agent
  2: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRole
    metadata:
      name: RELEASE-NAME-ngrok-operator-agent-role
    rules:
      - apiGroups:
          - ingress.k8s.ngrok.com
        resources:
          - tunnels
        verbs:
          - get
          - list
          - watch
          - patch
          - update
      - apiGroups:
          - ingress.k8s.ngrok.com
        resources:
          - tunnels/finalizers
        verbs:
          - update
      - apiGroups:
          - ingress.k8s.ngrok.com
        resources:
          - tunnels/status
        verbs:
          - get
          - patch
          - update
      - apiGroups:
          - ngrok.k8s.ngrok.com
        resources:
          - agentendpoints
        verbs:
          - get
          - list
          - watch
          - patch
          - update
      - apiGroups:
          - ngrok.k8s.ngrok.com
        resources:
          - agentendpoints/finalizers
        verbs:
          - update
      - apiGroups:
          - ngrok.k8s.ngrok.com
        resources:
          - agentendpoints/status
        verbs:
          - get
          - patch
          - update
      - apiGroups:
          - ngrok.k8s.ngrok.com
        resources:
          - ngroktrafficpolicies
        verbs:
          - get
          - list
          - watch
      - apiGroups:
          - ingress.k8s.ngrok.com
        resources:
          - domains
        verbs:
          - get
          - list
          - watch
          - create
          - patch
          - delete
      - apiGroups:
          - ""
        resources:
          - events
        verbs:
          - create
          - patch
  3: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRoleBinding
    metadata:
      name: RELEASE-NAME-ngrok-operator-agent-rolebinding
    roleRef:
      apiGroup: rbac.authorization.k8s.io
      kind: ClusterRole
      name: RELEASE-NAME-ngrok-operator-agent-role
    subjects:
      - kind: ServiceAccount
        name: RELEASE-NAME-ngrok-operator-agent
        namespace: NAMESPACE
